### PR TITLE
Added XAI features.

### PR DESCRIPTION
**Pull request by nikleet**
Initial commit of XAI features.

### CODE REVIEW

File: rbm_models/clust_dbn.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 import sys
 
 import numpy as np
+import pandas as pd
```
Comment (nikleet) [inline rbm_models/clust_dbn.py:6]: This import is no longer necessary

File: rbm_models/clust_dbn.py [hunk 2]
```diff
@@ -93,8 +94,86 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
...
+        y = torch.as_tensor(self.scaler.transform(y), dtype = dt)
+        y = self.fc.forward(y)
+        if isinstance(y,list) or isinstance(y,tuple):
+            y = y[0]
+        y = y.detach().cpu().numpy()
+        
+        return y
+    
+    
+    def numpy_forward_image(self, x: np.ndarray) -> np.ndarray:
```
Comment (nikleet) [inline rbm_models/clust_dbn.py:135]: This function isn't completely necessary at the moment either.

Comment (nlahaye) [reply rbm_models/clust_dbn.py:135]: Are these transformations from numpy to torch something we can do outside of the ClustDBN class - maybe in the xai.py script prior to passing the data through the models?

**Review by nlahaye — COMMENTED** [review]
These additions look great -I just had one comment/discussion point on the numpy <-> torch transformation locations. Once we resolve that I think we can add all these in.

**Review by nlahaye — APPROVED** [review]

**Pull request merged**